types/office-js-preview/index.d.ts

```diff
@@ -525,14 +525,55 @@ declare namespace Office {
+         * To learn how to specify a valid action ID and a key combination, see {@link https://docs.microsoft.com/office/dev/add-ins/design/keyboard-shortcuts | Add custom keyboard shortcuts to your Office Add-ins}. (Note tht a key combination can be `null`, in which case, the action keeps the key combination specified in the JSON file.)
+         * @returns A promise that resolves when every custom shortcut assignment in `shortcuts` has been registered. Even if there is a conflict with existing shortcuts, the customized shortcut will be registered;
+         * otherwise, the promise will be rejected with error code and error message; an error code "InvalidOperation" is returned if any action ID in `shortcuts` does not exist, or if shortcut combination is invalid.
+         */
+        replaceShortcuts(shortcuts: {[actionId:string]:string}): Promise<void>;   
+        /**
+         * Gets the existing shortcuts for the add-in. The set always includes (1) the shortcuts defined in the add-in's extended manifest for keyboard shortcuts and (2) the current user's custom shortcuts if those exist.
+         * The shortcut can be `null` if it conflicts with the shortcut of another add-in or with the Office application. specifically, it would be `null` if, when prompted to choose which shortcut to use, the user didn't choose the action of the current add-in. For more information about conflicts with shortcuts, see  {@link https://docs.microsoft.com/office/dev/add-ins/design/keyboard-shortcuts#avoid-key-combinations-in-use-by-other-add-ins | Avoid key combinations in use by other add-ins}.
+         *
+         * @remarks
+         *
+         * **Requirement set**: KeyboardShortcuts 1.1
+         * @beta
+         * 
+         * @returns A promise that resolves to an object of shortcuts, with keys being the IDs of the actions (as defined in an extended manifest) and values being the shortcut combinations; for example, {"SetItalic": "Ctrl+1", "SetBold": "Ctrl+2", "SetUnderline": null}.
+         */
+        getShortcuts(): Promise<{[actionId:string]:string|null}>;   
+        /**
+         * Checks if a set of shortcut combinations are currently in use for the user, as defined by another add-in or by the host Office application.
+         *
+         * @remarks
+         *
+         * **Requirement set**: KeyboardShortcuts 1.1
+         * @beta
+         * 
+         * @param shortcuts An array of shortcut combinations; e.g., ["Ctrl+1", "Ctrl+2"].
+         * @returns A promise that resolves to an array of objects. Each object consists of a shortcut combination and Boolean value. The value is `true` if the shortcut combination conflicts with a shortcut of another add-ins or with a shortcut of the Office application; otherwise, `false`. For example, `[{shortcut:"Ctrl+1", inUse:true},{shortcut:"Ctrl+2", inUse:false}]`.
```

Comment:
```suggestion
         * @returns A promise that resolves to an array of objects. Each object consists of a shortcut combination and Boolean value. The value is `true` if the shortcut combination conflicts with a shortcut of another add-in or with a shortcut of the Office application; otherwise, `false`. For example, `[{shortcut:"Ctrl+1", inUse:true},{shortcut:"Ctrl+2", inUse:false}]`.
```